7a_nwis_munge/src/munge_nyc_dep_reservoir_data.R

```diff
@@ -0,0 +1,61 @@
+
+munge_nyc_dep_temperature <- function(in_ind, out_ind, xwalk) {
+
+
+  dat <- readxl::read_xlsx(in_ind) %>%
+    # Removing unnecessary columns.
+    dplyr::select(-c(`Surface Elevation (ft.)`, `Thermocline (m)`, `Profile Id`,
+                     Analyte, Units, Qualifier)) %>%
+    filter(!is.na(Value)) %>% # Filtering the NA temperature values.
+    mutate(date =  as.Date(`Sample Date`), # Creating date column.
+           site_id = as.character(xwalk[Reservoir])) %>%
+    rename(source_id = Site, dateTime = `Sample Date`,
+           depth = `Depth (m)`, temp = Value)
+
+  # Filtering the sites near the dam (start with "1")
+  cannonsville_dam_site <- filter(dat, source_id %in% '1WDC')
+  pepacton_dam_site <- filter(dat, source_id %in% '1EDP')
+
+  # Selecting the unique dates associated with the sites near dam.
+  cannonsville_dates <- unique(cannonsville_dam_site$date)
+  pepacton_dates <- unique(pepacton_dam_site$date)
+
+  # Filtering out the dam sites and dates associated with them.
+  dates_count <- dat %>%
+    filter(!source_id %in% '1WDC' & !date %in% cannonsville_dates &
+             !source_id %in% '1EDP' & !date %in% pepacton_dates) %>%
+    # Grouping by reservoir and site to get the dates of measurement per reservoir/ site.
+    group_by(site_id, source_id) %>%
+    mutate(n_dates = length(unique(date))) %>%
+    ungroup()
+
+  # Finding the site with the maximum number of days with observation (n_dates).
+  max_dates <- dates_count %>%
+    group_by(site_id, date) %>%
+    filter(n_dates == max(n_dates)) %>%
+    ungroup()
+
+  # Selecting the site with maximum n_dates.
+  sites_max_dates <- unique(max_dates$source_id)
+
+  # Filtering the data to one profile per day per reservoir per site.
+  daily_profile <- dat %>%
+    # Filtering to the sites with the maximum n_dates
+    filter(source_id %in% sites_max_dates) %>%
```

Comment:
this is filtering to all sites that had the maximum number of observations on any given data, which will keep _all_ of that site's observations, not just those dates.